dashboard/test/models/unit_test.rb

```diff
@@ -2111,6 +2111,39 @@ class CanViewVersion < ActiveSupport::TestCase
     assert_equal "test-localized-title-default", unit.localized_title
   end
 
+  test 'next_unit returns next unit if there is another unit in unit group' do
+    unit_group = create :unit_group
+    unit1 = create :unit
+    unit2 = create :unit
+    create :unit_group_unit, unit_group: unit_group, script: unit1, position: 1
+    create :unit_group_unit, unit_group: unit_group, script: unit2, position: 2
+    unit1.reload
+
+    student = create :student
+
+    assert_equal unit2, unit1.next_unit(student)
+  end
+
+  test 'next_unit returns nil if there is no next unit in unit group' do
+    unit1 = create :unit
+    unit2 = create :unit
+    unit_group = create :unit_group
+    create :unit_group_unit, unit_group: unit_group, script: unit1, position: 1
+    create :unit_group_unit, unit_group: unit_group, script: unit2, position: 2
+
+    student = create :student
+
+    assert_nil unit2.next_unit(student)
+  end
```

Comment:
How are the units associated with the given student user here?

Reply:
A user is needed for the `units_for_user` function. I didn't write tests for cases where some users can see a unit and some can't but I'm happy to if you think it's necessary. I think the only scenario that happens in these days is an `in_development` unit within a launched unit group.

Reply:
Gotcha thank you! No that doesn't seem necessary, just wanted to make sure I understood how the `user` was involved